src/demos/demo1.ts

```diff
@@ -27,6 +29,19 @@ function main() {
       },
     },
   });
+
+  // DEMO - Add an angled wall to the scene
+  const angleWall = new SimWall(this.scene, this.world, {
```

Comment:
This part I'm a bit stuck on - the SimObjects require these two arguments. While these objects will definitely need references to these, I'm wondering what the best approach is to get those references.

Here is some ideas: 
1. change it to SimObject(simulator: Sim3D) and pass in the simulator reference. Simulator can then expose the scene and world.
2. Disallow making SimObjects directly and expose a set of functions on Sim3D to construct these objects
3. Make Simulator a singleton that can be accessed from anywhere. Most likely there would be only 1 Simulator at any time so this might be okay and simple to implement.

Reply:
I'm kind of flip-flopping between 1 and 2.

1 seems the cleanest, and also makes it clear that a particular SimObject is bound to a particular simulator instance. Downside is that we end up leaking the world and scene as part of the publicly accessible interface.

For 2, we can expose generalized functions e.g. addRobot(), addObject(), and have those functions essentially act as factories, and generate the appropriate objects dynamically. Downside is that the spec will have to be very detailed (although we could make liberal use of optional fields). But, it'll be a very clear definition of what constitutes a robot, or game object, and could play nicely into us building a custom interface for configuring a robot.

Reply:
Gave it a little bit more thought, and I'm liking option 2 more. We could have this implemented on Sim3D:

`public addGameObject(spec: IGameObjectSpec): ISimObjectHandle`

where the `IGameObjectSpec` can be used to describe the type of object you want to add (basic shapes, etc). and `ISimObjectRef` will look something like:

```typescript
interface ISimObjectRef {
    guid: string;
    type: string; 
}
```

This way, SimObjects are opaque to the simulator. Then we can have corresponding `Handles` that take in a `ISimObjectRef` and a Sim3D instance. We can expose a method on Sim3D to take in a `ISimObjectRef` and return a raw SimObject. The handle can then interact with the SimObject opaquely